INTITA/Modules/LogInScreen/LogInViewController.swift

```diff
@@ -90,83 +101,117 @@ class LogInViewController: UIViewController, Storyboarded, AlertAcceptable {
 }
 
 extension LogInViewController: UITableViewDelegate, UITableViewDataSource {
+    
     func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return 5
+        return 7
     }
     
     func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
-        switch indexPath.row {
-        case 0:
-            return view.frame.height / 3
-        case 1, 2:
-            return view.frame.height / 9
-        case 3:
-            return 25
-        case 4:
-            return 100
+        
+        let nameCell = LoginCells(rawValue: indexPath.row)
+        var heightCell: CGFloat
+        
+        switch nameCell {
+        case .logoImageCell:
+            heightCell = 243
+        case .emptyCellOne:
+            heightCell = 52
+        case .loginButtonCell, .emailTextFieldCell:
+            heightCell = 78
+        case .passwordTextFieldCell:
+            heightCell =  57
+        case .linksButtonCell:
+            heightCell = 40
         default:
-            return 125.0
+            heightCell = 20.0
         }
+        return heightCell
     }
     
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         
-        switch indexPath.row {
-        case 0:
+        let nameCell = LoginCells(rawValue: indexPath.row)
+        var cell: UITableViewCell?
+        
+        switch nameCell {
+        case .logoImageCell:
             guard let logoCell = tableView.dequeueReusableCell(withIdentifier: "reuseForLogo") as? LogoTableViewCell else { return UITableViewCell() }
```

Comment:
you don't need guards now... you have return in the end of the function

Reply:
use optional chaining instead

Reply:
It has done. You was right. Thanks for a pies of advice.